File: AdminPanel/AdminPanel.Web/Modules/ECommerceWebsite/Carts/CartsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel.ECommerceWebsite.Columns;

[ColumnsScript("ECommerceWebsite.Carts")]
[BasedOnRow(typeof(CartsRow), CheckNames = true)]
public class CartsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string UserId { get; set; }
    public string ProductName { get; set; }
    public int Quantity { get; set; }
}

File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/ECommerceWebsite/CartsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CartsRow } from "./CartsRow";

export interface CartsColumns {
    Id: Column<CartsRow>;
    UserId: Column<CartsRow>;
    ProductName: Column<CartsRow>;
    Quantity: Column<CartsRow>;
}

export class CartsColumns extends ColumnsBase<CartsRow> {
    static readonly columnsKey = 'ECommerceWebsite.Carts';
    static readonly Fields = fieldsProxy<CartsColumns>();
}